### PR TITLE
feat(tags): add locale as optional parameter

### PR DESCRIPTION
**Pull request by Chroq**
Hi, 

Just few changes to adapt function signature to have access to the locale parameter used by .
I really need to have the possibility to set this parameter.

### CODE REVIEW

**Comment by freekmurze** [comment]
Please also add tests and update the readme.

**Comment by Chroq** [comment]
Thank you for the feedback !
I had tests and doc

File: src/HasTags.php [hunk 1]
```diff
@@ -73,12 +73,13 @@ public function setTagsAttribute($tags)
     /**
      * @param \Illuminate\Database\Eloquent\Builder $query
      * @param array|\ArrayAccess|\Spatie\Tags\Tag $tags
+     * @param null $locale
      *
      * @return \Illuminate\Database\Eloquent\Builder
      */
-    public function scopeWithAllTags(Builder $query, $tags, string $type = null): Builder
+    public function scopeWithAllTags(Builder $query, $tags, string $type = null, $locale = null): Builder
```
Comment (freekmurze) [inline src/HasTags.php:80]: You can add a `string` type hint to the `locale` parameter

File: src/HasTags.php [hunk 2]
```diff
@@ -119,9 +127,15 @@ public function scopeWithAllTagsOfAnyType(Builder $query, $tags): Builder
         return $query;
     }
 
-    public function scopeWithAnyTagsOfAnyType(Builder $query, $tags): Builder
+    /**
+     * @param Builder $query
```
Comment (freekmurze) [inline src/HasTags.php:131]: Please type everything you can and use fully qualified class name in this docblock (like the existing docblocks)

File: tests/HasTagsScopesTest.php [hunk 3]
```diff
@@ -115,4 +122,36 @@ public function it_provides_as_scope_to_get_all_models_that_have_all_of_the_give
 
         $this->assertEquals(['model5'], $testModels->pluck('name')->toArray());
     }
+
+    /** @test */
+    public function it_provides_as_scope_to_get_all_models_that_have_all_of_the_given_tags_with_type_and_a_specific_locale()
+    {
+        $testModels = TestModel::withAllTags(['tagG'], 'translatedTag' ,'fr')->get();
+
+        $this->assertEquals(['model7'], $testModels->pluck('name')->toArray());
```
Comment (freekmurze) [inline tests/HasTagsScopesTest.php:131]: Make this more clear by seeding tags in two different locales and assets that for each locale you get the right string back.

File: tests/HasTagsScopesTest.php [hunk 4]
```diff
@@ -115,4 +122,36 @@ public function it_provides_as_scope_to_get_all_models_that_have_all_of_the_give
...
+    }
+
+    /** @test */
+    public function it_provides_as_scope_to_get_all_models_that_have_all_of_the_given_tags_with_any_type_and_a_specific_locale()
+    {
+        $testModels = TestModel::withAllTagsOfAnyType(['tagG'], 'fr')->get();
+
+        $this->assertEquals(['model7'], $testModels->pluck('name')->toArray());
+    }
```
Comment (freekmurze) [inline tests/HasTagsScopesTest.php:156]: Also add tests for the case where you try to get a locale, but no locale is used for a tag. What happens than?

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: src/HasTags.php (hunk 1 above)
Comment (Chroq) [reply src/HasTags.php]: Fixed, thank you

File: src/HasTags.php (hunk 2 above)
Comment (Chroq) [reply src/HasTags.php]: Fixed, thank you

File: tests/HasTagsScopesTest.php (hunk 3 above)
Comment (Chroq) [reply tests/HasTagsScopesTest.php]: Fixed, thank you

File: tests/HasTagsScopesTest.php (hunk 4 above)
Comment (Chroq) [reply tests/HasTagsScopesTest.php:186]: I added some tests but I'm not sure it doesn't duplicate the existing ones.

**Comment by freekmurze** [comment]
After thinking it through, I have decided not to merge this in because of the added complexity for something that I don't need for our projects. 

But your work is not lost, you can use it in a fork, or in the project where you need this.

**Comment by Chroq** [comment]
Ok, I understand. I'll overload the HasTag trait in my project.
I appreciate your concern for my PR. 

Have a nice day.

**Pull request closed without merging**